cpu/kinetis/periph/timer.c

```diff
@@ -720,10 +720,22 @@ unsigned int timer_read(tim_t dev)
     /* demultiplex to handle two types of hardware timers */
     switch (_timer_variant(dev)) {
         case TIMER_PIT:
-            return pit_read(_pit_index(dev));
+            if(PIT_CLK_IS_EN()) {
+                return pit_read(_pit_index(dev));
+            }
+            /* uninitialized timer*/
+            else {
```

Comment:
You don't need the `else`. Here and in the TIMER_LPTMR case as well.

Maybe this could be even more simplified with just one `return 0;` and the end if the function and the of `break` at the of each cases in the switch (including the `default`).

Reply:
If I do that I get:

```
/home/francisco/workspace/RIOT/cpu/kinetis/periph/timer.c:711:16: error: this statement may fall through [-Werror=implicit-fallthrough=]
             if (PIT_CLK_IS_EN()) {
                ^
/home/francisco/workspace/RIOT/cpu/kinetis/periph/timer.c:715:9: note: here
         case TIMER_LPTMR:
```